lib/teiserver/protocols/spring/spring_in.ex

```diff
@@ -1359,10 +1359,33 @@ defmodule Teiserver.Protocols.SpringIn do
     state
   end
 
-  defp do_handle("RING", username, _msg_id, state) do
-    userid = CacheUser.get_userid(username)
-    CacheUser.ring(userid, state.userid)
-    state
+  # https://springrts.com/dl/LobbyProtocol/ProtocolDescription.html#RING:client
+  # extended to accept originator as a second argument (only allowed when sent by bots)
+  # this allows spads to inform clients who originally rang them with spads-command !ring
+  defp do_handle("RING", data, _msg_id, state) do
+    case Regex.run(~r/(\S+) (\S+)/, data) do
```

Comment:
somewhat nitpick, for such a simple regex you could also pattern match on a split:
```elixir
case String.split(data) do
  [_, originator] -> 
    if ... do
      # new version
    end
  _ -> # fallback
end
```

I don't think it matters much since it's not in any hotpath, so the cost of compiling the regex can be ~ignored.

Reply:
Should be fixed